### PR TITLE
[EDMTP-248] Add more options to @Select

### PR DESCRIPTION
**Pull request by ldzmitrenka**

### CODE REVIEW

File: aem-authoring-toolkit-api/src/main/java/com/exadel/aem/toolkit/api/annotations/widgets/select/Select.java [hunk 1]
```diff
@@ -37,39 +37,101 @@
...
+     * @return true or false
+     */
+    boolean multiple() default false;
+
+    /**
+     * Returns true to translate the options, false otherwise.
+     *
+     * @return true or false
+     */
+    boolean translateOptions() default false;
```
Comment (liubou-masiuk) [inline aem-authoring-toolkit-api/src/main/java/com/exadel/aem/toolkit/api/annotations/widgets/select/Select.java:94]: `translateOptions` is true by default in the granite component, so it also should be true here

File: aem-authoring-toolkit-api/src/main/java/com/exadel/aem/toolkit/api/annotations/widgets/select/Select.java [hunk 2]
```diff
@@ -37,39 +37,101 @@
...
+     * @return true or false
+     */
+    boolean emptyOption() default false;
+
+    /**
+     * Return default or quite value.
+     *
+     * @return String value
+     */
+    String variant() default "";
```
Comment (liubou-masiuk) [inline aem-authoring-toolkit-api/src/main/java/com/exadel/aem/toolkit/api/annotations/widgets/select/Select.java:119]: there are 2 possible options for `variant`: `default` and `quiet`. I think they should be represented by an enum

File: aem-authoring-toolkit-api/src/main/java/com/exadel/aem/toolkit/api/annotations/widgets/select/Select.java [hunk 3]
```diff
@@ -37,39 +37,101 @@
...
+
+    /**
+     * Return true to generate the SlingPostServlet @Delete hidden input based on the
+     * field name.
+     *
+     * @return true or false
+     */
+    boolean deleteHint() default false;
```
Comment (liubou-masiuk) [inline aem-authoring-toolkit-api/src/main/java/com/exadel/aem/toolkit/api/annotations/widgets/select/Select.java:127]: `deleteHint` is true by default in the granite component, so it also should be true here

File: aem-authoring-toolkit-api/src/main/java/com/exadel/aem/toolkit/api/annotations/widgets/select/Select.java [hunk 4]
```diff
@@ -37,39 +34,103 @@
...
+
+    /**
+     * Indicates if the user is able to select multiple selections.
+     *
+     * @return true or false
+     */
+    boolean multiple() default false;
+
+    /**
+     * Returns true to translate the options, false otherwise.
```
Comment (smiakchilo) [inline aem-authoring-toolkit-api/src/main/java/com/exadel/aem/toolkit/api/annotations/widgets/select/Select.java:87]: (cosmetic) I would say, "If set to true, the options' labels are translated by AEM". Not exactly different in meaning, but a little bit of variety)

File: aem-authoring-toolkit-api/src/main/java/com/exadel/aem/toolkit/api/annotations/widgets/select/Select.java [hunk 5]
```diff
@@ -37,39 +34,103 @@
...
+     *
+     * @return true or false
+     */
+    boolean translateOptions() default true;
+
+    /**
+     * Returns true to sort the options based on the text, false otherwise.
```
Comment (smiakchilo) [inline aem-authoring-toolkit-api/src/main/java/com/exadel/aem/toolkit/api/annotations/widgets/select/Select.java:94]: "If set to true, the options are sorted according to their labels' alphabetical order"

File: aem-authoring-toolkit-api/src/main/java/com/exadel/aem/toolkit/api/annotations/widgets/select/Select.java [hunk 6]
```diff
@@ -37,39 +34,103 @@
...
+    /**
+     * Return true to generate the SlingPostServlet @Delete hidden input based on the
+     * field name.
+     *
+     * @return true or false
+     */
+    boolean deleteHint() default true;
+
+    /**
+     * Return true to generate the SlingPostServlet @Delete hidden input based on the
```
Comment (smiakchilo) [inline aem-authoring-toolkit-api/src/main/java/com/exadel/aem/toolkit/api/annotations/widgets/select/Select.java:129]: The javadoc for prev item has been occasionally duplicated, seemingly

File: aem-authoring-toolkit-api/src/main/java/com/exadel/aem/toolkit/api/annotations/widgets/select/SelectVariant.java [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.exadel.aem.toolkit.api.annotations.widgets.select;
+
+/**
+ * Contains possible values of {@link Select#variant()} property
```
Comment (smiakchilo) [inline aem-authoring-toolkit-api/src/main/java/com/exadel/aem/toolkit/api/annotations/widgets/select/SelectVariant.java:17]: Circular reference detected. See `Select.java` line 112

**Pull request merged**